rotkehlchen/chain/ethereum/decoding/base.py

```diff
@@ -130,7 +136,18 @@ def decode_erc20_721_transfer(
             else:
                 notes = f'{verb} {amount} {token.symbol} from {counterparty} to {location_label}'
         elif token.token_kind == EvmTokenKind.ERC721:
-            token_id = hex_or_bytes_to_int(tx_log.topics[3])
+            try:
+                if token.evm_address in NAUGHTY_ERC721:  # id is in the data
+                    token_id = hex_or_bytes_to_int(tx_log.data[0:32])
```

Comment:
I wrote the negation first since was the most probable case based on my believes related to branch prediction but after a research seems that python doesn't benefit from this and recent processors neither

Reply:
compilers are not stupid. You write for humans and not for compilers. And negating everyting is hard to read